dot/parachain/types/async_backing.go

```diff
@@ -17,3 +22,90 @@ type AsyncBackingParams struct {
 	// When async backing is disabled, the only valid value is 0.
 	AllowedAncestryLen uint32 `scale:"2"`
 }
+
+// InboundHrmpLimitations constraints on inbound HRMP channels.
+type InboundHrmpLimitations struct {
+	// An exhaustive set of all valid watermarks, sorted ascending.
+	//
+	// It's only expected to contain block numbers at which messages were
+	// previously sent to a para, excluding most recent head.
+	ValidWatermarks []uint
+}
+
+// OutboundHrmpChannelLimitations constraints on outbound HRMP channels.
+type OutboundHrmpChannelLimitations struct {
+	// The maximum bytes that can be written to the channel.
+	BytesRemaining uint32
+	// The maximum messages that can be written to the channel.
+	MessagesRemaining uint32
+}
+
+// Constraints on the actions that can be taken by a new parachain block. These
+// limitations are implicitly associated with some particular parachain, which should
+// be apparent from usage.
+type Constraints struct {
```

Comment:
Let's move this type to the inclusion simulator file.

This package contains types that we need in multiple subsystems.
Because we will never use this in other subsystems, we only use this in prospective parachain.

Move other types you added to this file that are not used in multiple subsystems.